packages/starlight/utils/translations.ts

```diff
@@ -4,7 +4,9 @@ import type { i18nSchemaOutput } from '../schemas/i18n';
 import { createTranslationSystem } from './createTranslationSystem';
 
 type UserI18nSchema = 'i18n' extends DataCollectionKey
-	? CollectionEntry<'i18n'>['data']
+	? CollectionEntry<'i18n'> extends { data: infer T }
```

Comment:
This new change is required to due this [change](https://github.com/withastro/astro/pull/10001) in Astro `v4.4.0` which makes CC types include collection names even when a matching directory is absent.